src/integrationTest/java/com/mongodb/spark/sql/connector/read/MongoSparkConnectorReadTest.java

```diff
@@ -442,6 +443,26 @@ void testReadsAreSupportedWithFilters() {
             .collectAsList());
   }
 
+  @Test
+  void testCustomMongoClientFactory() {
+    SparkSession spark = getOrCreateSparkSession();
+
+    List<BsonDocument> collectionData =
+        toBsonDocuments(spark.read().textFile(READ_RESOURCES_HOBBITS_JSON_PATH));
+    getCollection().insertMany(collectionData);
+
+    spark
+        .read()
+        .format("mongodb")
+        .option(
+            ReadConfig.READ_PREFIX + ReadConfig.CLIENT_FACTORY_CONFIG,
```

Comment:
Here we test setting an option that previously would not have been picked up after the ReadConfig options had been specialized due to it being converted to lowercase.

Reply:
Ack